### PR TITLE
PDF Manipulation API

### PR DESCRIPTION
**Pull request by kadamidev**
Closes https://github.com/desci-labs/nodes-web-v2/issues/689

## Description of the Problem / Feature
- Prepublication needs a service to create a PDF ready for distribution
## Explanation of the solution
- [x] Create an API in the isolated media server to handle PDF manipulation
- [x] Basic template with the required elements for the cover page
- [x] Nodes backend API wrapper that calls the generation API, and maintains results in the db. (Partially complete, will need changes once DOIs are implemented)
- [ ] DOI implementation
- [x] Final copy implementation

## Instructions on making this work
- Contains DB migrations

### CODE REVIEW

**Comment by carlaost** [comment]
@kadamidev i got the copy from philipp - https://docs.google.com/document/d/1iPectaHsfZ-RrTfF10OzBOvR5RcNxAnsHxL9aZkRzGk/edit?usp=sharing
double-checking w him if he wants any special design or to literally implement this doc

**Comment by carlaost** [comment]
@kadamidev find the final design for submission manuscript cover page here: https://www.figma.com/design/SGc8831RAsh4ZZ8gKQgX0y/DeSci-Nodes-v2?node-id=2783-50796&t=1wPNavMUge7iev2U-4
note that only the following three badges should be shown on the title page if claimed: open data, open code, reproducibility enabled.

**Comment by carlaost** [comment]
@kadamidev another question from philipp: is there a way to make the title text container sensitive to natural breaks like ":" or "-"? these are often used to separate title from subtitle and it would be more polished to insert a linebreak there. it comes with considerations like don't break line if that only leaves one or two words on the preceding line i guess but just generally, is that something you think is reasonable?

File: .vscode/launch.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
       "type": "node",
       "request": "attach",
       "restart": true,
-      "port": 9229,
```
Comment (hubsmoke) [inline .vscode/launch.json:12]: was there an issue requiring port change?

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -203,11 +204,14 @@ services:
...
       target: dev
     container_name: "media_isolated"
+    extra_hosts:
+      - host.docker.internal:host-gateway
     volumes:
       - ./desci-media-isolated:/usr/src/app
     ports:
       - "9777:9777" # debugger
-      # - "7771:7771" # Uncomment if you want to test the media server from the host machine
+      - "7771:7771" # Uncomment if you want to test the media server from the host machine
```
Comment (hubsmoke) [inline docker-compose.dev.yml:213]: reason to check in uncommented?

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -203,11 +204,14 @@ services:
       context: ./desci-media-isolated
       target: dev
     container_name: "media_isolated"
+    extra_hosts:
```
Comment (hubsmoke) [inline docker-compose.dev.yml:207]: what was this fixing?

**Review by hubsmoke — APPROVED** [review]
Can you address my comments? @kadamidev

**Pull request merged**